pkg/client/client.go

```diff
@@ -56,38 +57,41 @@ func newUnixSocketClient(ctx context.Context, config *Config) (*Client, error) {
 	}, nil
 }
 
-func newNetworkClient(ctx context.Context, config *Config) (*Client, error) {
+func newNetworkClient(ctx context.Context, config *Config) (*Client, error) {  
 	certificate, err := tls.LoadX509KeyPair(
 		config.CertFile,
 		config.KeyFile,
 	)
 	if err != nil {
 		return nil, fmt.Errorf("error loading the X.509 key pair: %v", err)
 	}
-
-	certPool := x509.NewCertPool()
-	rootCA, err := ioutil.ReadFile(config.CARootFile)
-	if err != nil {
-		return nil, fmt.Errorf("error reading the CA Root file certificate: %v", err)
+	tlsConfig := tls.Config {
+		ServerName:   config.Hostname,
+		Certificates: []tls.Certificate{certificate},
+		InsecureSkipVerify: config.InsecureSkipMutualTLSAuth,
 	}
-
-	ok := certPool.AppendCertsFromPEM(rootCA)
-	if !ok {
-		return nil, fmt.Errorf("error appending the root CA to the certificate pool")
+	certPool := x509.NewCertPool()
+	if(!config.InsecureSkipMutualTLSAuth){
```

Comment:
To keep the spacing coherent.
```suggestion
	if !config.InsecureSkipMutualTLSAuth {
```